src/approaches/grammar_search_invention_approach.py

```diff
@@ -558,6 +564,47 @@ def _evaluate_with_operators(self, predicates: FrozenSet[Predicate],
         return CFG.grammar_search_bf_weight * total_branching_factor
 
 
+@dataclass(frozen=True, eq=False, repr=False)
+class _TaskPlanningHeuristic(_OperatorLearningBasedHeuristic):
+    """Score a predicate set by learning operators and planning in the training
+    tasks. The score corresponds to the total number of nodes expanded across
+    all training problems. If no plan is found, a large penalty is added, which
+    is meant to be an upper bound on the number of nodes that could be expanded.
+    """
+
+    def _evaluate_with_operators(self, predicates: FrozenSet[Predicate],
+                                 pruned_atom_data: List[GroundAtomTrajectory],
+                                 segments: List[Segment],
+                                 strips_ops: List[STRIPSOperator],
+                                 option_specs: List[OptionSpec]) -> float:
+        del pruned_atom_data, segments  # unused
+        score = 0.0
+        node_expansion_upper_bound = 1e7
+        option_model = create_option_model("default", lambda s, a: s)  # dummy
```

Comment:
Let's just make a `DummyOptionModel = ...` in option_model.py and pass that directly into sesame_plan below?